### PR TITLE
Tinkerbell generate worker nodes capi objects

### PR DESCRIPTION
**Pull request by tatlat**
*Issue #, if available:*

*Description of changes:*
Implemented code to generate worker nodes CAPI objects for Tinkerbell provider

*Testing (if applicable):*
unit tests

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: pkg/providers/tinkerbell/workers.go [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+	wcc := spec.Cluster.Spec.WorkerNodeGroupConfigurations
+	workerTemplateNames := make(map[string]string, len(wcc))
+	kubeadmTemplateNames := make(map[string]string, len(wcc))
+
+	for _, wc := range wcc {
+		workerTemplateNames[wc.Name] = clusterapi.WorkerMachineTemplateName(spec, wc)
+		kubeadmTemplateNames[wc.Name] = clusterapi.DefaultKubeadmConfigTemplateName(spec, wc)
+	}
+
+	TemplateBuilder, err := generateTemplateBuilder(spec)
```
Comment (cxbrowne1207) [inline pkg/providers/tinkerbell/workers.go:36]:
```suggestion
	templateBuilder, err := generateTemplateBuilder(spec)
```

File: pkg/providers/tinkerbell/workers.go [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+	if err != nil {
+		return nil, err
+	}
+
+	workersYaml, err := TemplateBuilder.GenerateCAPISpecWorkers(spec, workerTemplateNames, kubeadmTemplateNames)
```
Comment (cxbrowne1207) [inline pkg/providers/tinkerbell/workers.go:41]:
```suggestion
	workersYaml, err := templateBuilder.GenerateCAPISpecWorkers(spec, workerTemplateNames, kubeadmTemplateNames)
```

File: pkg/providers/tinkerbell/workers.go [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+	kubeadmTemplateNames := make(map[string]string, len(wcc))
+
+	for _, wc := range wcc {
+		workerTemplateNames[wc.Name] = clusterapi.WorkerMachineTemplateName(spec, wc)
+		kubeadmTemplateNames[wc.Name] = clusterapi.DefaultKubeadmConfigTemplateName(spec, wc)
+	}
+
+	TemplateBuilder, err := generateTemplateBuilder(spec)
+	if err != nil {
+		return nil, err
```
Comment (cxbrowne1207) [inline pkg/providers/tinkerbell/workers.go:38]:
```suggestion
		return nil, errors.Wrap(err, "generating tinkerbell template builder")

```

File: pkg/providers/tinkerbell/workers.go [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+	}
+
+	workersYaml, err := TemplateBuilder.GenerateCAPISpecWorkers(spec, workerTemplateNames, kubeadmTemplateNames)
+	if err != nil {
+		return nil, err
```
Comment (cxbrowne1207) [inline pkg/providers/tinkerbell/workers.go:43]:
```suggestion
		return nil, errors.Wrap(err, "generating tinkerbell workers yaml spec")
```

File: pkg/providers/tinkerbell/workers.go [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+	// Workers represents the Tinkerbell specific CAPI spec for worker nodes.
+	Workers        = clusterapi.Workers[*tinkerbellv1.TinkerbellMachineTemplate]
+	workersBuilder = capiyaml.WorkersBuilder[*tinkerbellv1.TinkerbellMachineTemplate]
+)
+
+// WorkersSpec generates a Tinkerbell specific CAPI spec for an eks-a cluster worker nodes.
+// It talks to the cluster with a client to detect changes in immutable objects and generates new
+// names for them.
+func WorkersSpec(ctx context.Context, logger logr.Logger, client kubernetes.Client, spec *cluster.Spec) (*Workers, error) {
+	wcc := spec.Cluster.Spec.WorkerNodeGroupConfigurations
```
Comment (ddjjia) [inline pkg/providers/tinkerbell/workers.go:27]: Can we name it WorkerNodeGroupConfigurations, or wgc for short?

**Review by cxbrowne1207 — APPROVED** [review]

**Pull request merged**